### PR TITLE
Vasiliy linked list

### PR DESCRIPTION
**Pull request by vasiliypekhterev**
The implementation and unit tests of the Linked List were added.

### CODE REVIEW

File: vasiliy/src/main/java/data_structures/MyLinkedList.java [hunk 1]
```diff
@@ -1,6 +1,207 @@
...
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        return checkIndex(indexOf(o));
+    }
+
+    @Override
+    public boolean add(int index, Object o) {
+        if (!checkIndex(index)) {
```
Comment (gorobec) [inline vasiliy/src/main/java/data_structures/MyLinkedList.java:62]: You will duplicate this check in method:
private Node findNode(int index) {
        if (!checkIndex(index) || isEmpty()) {
            return null;
        }

File: vasiliy/src/main/java/data_structures/MyLinkedList.java [hunk 2]
```diff
@@ -1,6 +1,207 @@
...
+        Node node = head;
+        for (int i = 0; i < index; i++) {
+            node = node.getNext();
+        }
+        return node;
+    }
+
+    @Override
+    public Object get(int index) {
+        if (!checkIndex(index)) {
```
Comment (gorobec) [inline vasiliy/src/main/java/data_structures/MyLinkedList.java:102]: You will duplicate this check in method:
private Node findNode(int index) {
        if (!checkIndex(index) || isEmpty()) {
            return null;
        }

File: vasiliy/src/main/java/data_structures/MyLinkedList.java [hunk 3]
```diff
@@ -1,6 +1,207 @@
...
+        node.getPrevious().setNext(node.getNext());
+        node.getNext().setPrevious(node.getPrevious());
+        node.setNext(null);
+        node.setPrevious(null);
+        size--;
+    }
+
+    @Override
+    public Object remove(int index) {
+        if (!checkIndex(index)) {
```
Comment (gorobec) [inline vasiliy/src/main/java/data_structures/MyLinkedList.java:136]: You will duplicate this check in method:
private Node findNode(int index) {
        if (!checkIndex(index) || isEmpty()) {
            return null;
        }

File: vasiliy/src/main/java/data_structures/MyLinkedList.java [hunk 4]
```diff
@@ -1,6 +1,207 @@
...
+        if (node == null) {
+            return false;
+        }
+        removeNode(node);
+        return true;
+    }
+
+    @Override
+    public Object set(int index, Object o) {
+        if (!checkIndex(index)) {
```
Comment (gorobec) [inline vasiliy/src/main/java/data_structures/MyLinkedList.java:180]: You will duplicate this check in method:
private Node findNode(int index) {
        if (!checkIndex(index) || isEmpty()) {
            return null;
        }

File: vasiliy/src/main/java/data_structures/MyLinkedList.java [hunk 5]
```diff
@@ -1,6 +1,207 @@
...
+            clear();
+            return;
+        }
+        if (node == head) {
+            node.getNext().setPrevious(null);
+            head = node.getNext();
+            node.setNext(null);
+            return;
+        }
+        if (node == head) {
```
Comment (gorobec) [inline vasiliy/src/main/java/data_structures/MyLinkedList.java]: if (node == head) you will never reach this block, cause method will finish in previous block with the same statement

**Review by gorobec — CHANGES_REQUESTED** [review]
need work

**Comment by vasiliypekhterev** [comment]
The bugs were fixed.

**Comment by vasiliypekhterev** [comment]
The bugs were fixed.

**Pull request merged**